### ISSUE TITLE
Support jsonpath overrides

### ISSUE THREAD

**Issue by marun** [kubernetes-sigs/kubefed#520]
Overrides are easy to specify for simple paths (e.g. `spec.replicas`), but overriding paths with an indexed field (e.g. `spec.template.spec.containers[0].image`) requires using a separate configmap or secret resource to work around the lack of indexing support in the path-based mechanism.  This suggests supporting json-patch directives:

```yaml
apiVersion: generated.federation.k8s.io/v1alpha1
kind: FederatedDeployment
metadata:
  name: foo
  namespace: bar
spec:
  overrides:
  - clusterName: cluster1
     clusterOverrides:
     - path: spec.replicas
       value: 3
  - clusterName: cluster2
     # Support multiple directives and apply them in sequence
     jsonPatches:
     # Equivalent directive to the path-based override for cluster1
     - { "op": "replace", "path": "/spec/replicas", "value": 5 }
     # Indexed directive
     - { "op": "replace", "path": "/spec/template/spec/containers[0]/image", "value": "different-image" }  
```

It should be possible to supply both `clusterOverrides` and `jsonPatches` so long as the ordering of application is documented.

Previously: #400 

cc: @danehans @mvazquezc

**Comment by senyan** [comment]
support of jsonPatches would be a great step forward. In addition, I would feel if we could get more ``transformation`` logic would be even better while propagate to federated clusters. Please see [here](https://github.com/kubernetes-sigs/federation-v2/issues/546).

**Comment by senyan** [comment]
support of jsonPatches would be a great step forward. In addition, I would feel if we could get more ``transformation`` logic would be even better while propagate to federated clusters. Please see [here](https://github.com/kubernetes-sigs/federation-v2/issues/546).

**Comment by font** [comment]
Fixed by #991.

**Comment by senyan** [comment]
Amazing.
